src/agent/adu_core_interface/tests/adu_core_interface_ut.cpp

```diff
@@ -70,21 +71,21 @@ class ADUC_UT_ReportPropertyAsyncValues
     void
     set(ADUC_ClientHandle deviceHandleIn,
         const unsigned char* reportedStateIn,
-        size_t reportedStateLenIn,
+        size_t _reportedStateLenIn,
         IOTHUB_CLIENT_REPORTED_STATE_CALLBACK reportedStateCallbackIn,
         void* userContextCallbackIn)
     {
         deviceHandle = deviceHandleIn;
 
-        memcpy(reportedState, reportedStateIn, reportedStateLenIn);
+        reportedState = reinterpret_cast<const char*>(reportedStateIn);
+        reportedStateLen = reportedState.length();
 
-        reportedStateLen = reportedStateLenIn;
         reportedStateCallback = reportedStateCallbackIn;
         userContextCallback = userContextCallbackIn;
     }
 
     ADUC_ClientHandle deviceHandle{ nullptr };
-    unsigned char reportedState[1024]{}; /* The API treats this as a blob. */
+    std::string reportedState; /* The API treats this as a blob. */
```

Comment:
>blob [](http://example.com/codeflow?start=59&length=4)

see my other comment.  we can't have "this is treated as a blob" and "it's a std::string" -- they're contradictory

Reply:
good points.  will follow-up.